src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,33 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Could not add movie", e);
```

Comment:
Consider enhancing the error message in the `DataProcessingException` by including the movie's details or ID. This will help in identifying the specific movie that caused the error.